Common/OS.cs

```diff
@@ -160,5 +138,68 @@ public static decimal CpuUsage
                 { "LEAN Version", $"v{Globals.Version}"}
             };
         }
+
+        /// <summary>
+        /// Calculates the CPU usage in a background thread
+        /// </summary>
+        private class CpuPerformance : IDisposable
+        {
+            private readonly CancellationTokenSource _cancellationToken;
+            private readonly ManualResetEventSlim _exitEvent;
+            private readonly Task _cpuPerformanceTask;
+
+            /// <summary>
+            /// CPU usage as a percentage (0-100)
+            /// </summary>
+            public double CpuPercentage { get; private set; }
+
+            /// <summary>
+            /// Initializes an instance of the class and starts a new thread.
+            /// </summary>
+            public CpuPerformance()
+            {
+                _exitEvent = new ManualResetEventSlim();
+                _cancellationToken = new CancellationTokenSource();
+                _cpuPerformanceTask = Task.Factory.StartNew(CalculateCpu, null, _cancellationToken.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+            }
+
+            /// <summary>
+            /// Event loop that calculates the CPU percentage the process is using
+            /// </summary>
+            private void CalculateCpu(object _)
+            {
+                while (!_cancellationToken.IsCancellationRequested)
+                {
+                    var startTime = DateTime.UtcNow;
+                    var startCpuUsage = Process.GetCurrentProcess().TotalProcessorTime;
+                    var stopWatch = Stopwatch.StartNew();
+
+                    if (_exitEvent.Wait(1000))
+                    {
+                        return;
+                    }
+
+                    stopWatch.Stop();
+                    var endTime = DateTime.UtcNow;
+                    var endCpuUsage = Process.GetCurrentProcess().TotalProcessorTime;
+
+                    var cpuUsedMs = (endCpuUsage - startCpuUsage).TotalMilliseconds;
+                    var totalMsPassed = (endTime - startTime).TotalMilliseconds;
+                    var cpuUsageTotal = cpuUsedMs / totalMsPassed;
+
+                    CpuPercentage = cpuUsageTotal * 100;
```

Comment:
We will need some access control because written to the double isn't guaranteed to be atomic, meaning it can be read while being modified, see https://stackoverflow.com/a/11745701/7334720 https://stackoverflow.com/a/3677373/7334720 . Can use some interlock exchange for example 👍